app-creator/client/src/widgets/story-board/story-board.ts

```diff
@@ -102,49 +112,199 @@ export class Storyboard extends connect(store)(LitElement) {
+     */
+    if (state.eventType === EventType.changingPalette) {
+      store.dispatch(setEventType(EventType.none, true));
 
-      this.requestUpdate();
+      this.renderNewTemplate(template);
     }
   }
 
   /**
-   * Additional custom styles.
+   * Clears the value of the palette select input and restores initial state.
    */
-  @property({ type: Object }) styles = {};
+  clearSelectedPalette() {
+    if (this.paletteSelect) {
+      this.paletteSelect.value = '';
+    }
+    this.selectedPalette = Palette.light;
+    this.requestedPalette = Palette.light;
+  }
 
   /**
-   * Reference to storyboard element.
+   * Renders new template by calling generateUI and requesting update.
    */
-  @query(`#${STORYBOARD_ID}`) storyboard!: PaperCardElement;
+  renderNewTemplate(template: AppCreatorStore['template']) {
+    const { storyboard } = this;
+    if (storyboard) {
+      storyboard.innerHTML = ``;
+      generateUI(template, storyboard);
+    }
+
+    this.requestUpdate();
+  }
+
+  /**
+   * Sets the selected palette in state and opens the confirmation dialog.
+   */
+  handlePaletteChange(e: Event) {
+    this.requestedPalette = (e.target as HTMLSelectElement).value as Palette;
+    if (this.paletteConfimationDialog) {
+      this.paletteConfimationDialog.open();
+    }
+  }
+
+  /**
+   * Called when users confirm on the confirmation dialog when changing palettes.
+   */
+  paletteChangeConfirmation() {
+    this.selectedPalette = this.requestedPalette;
+    store.dispatch(setPalette(this.selectedPalette));
+    store.dispatch(setEventType(EventType.changingPalette, true));
+  }
 
   render() {
-    const { styles } = this;
+    const { styles, handlePaletteChange, paletteChangeConfirmation } = this;
 
     const isMobile =
       store.getState().template.config?.device === DeviceType.mobile;
 
+    const paletteSelectInput = html`
+      <div id="palette-select-container">
+        <select
+          name="Select palette"
+          id="select-palette"
+          @change=${handlePaletteChange}
+        >
+          <option value="" disabled selected>Change Palette</option>
+          ${Object.keys(Palette).map(
+            (palette) => html`<option value=${palette}>${palette}</option>`
+          )}
+        </select>
+      </div>
+    `;
+
+    const paletteChangeConfirmationDialog = html`
+      <paper-dialog with-backdrop id="palette-change-confirmation-dialog">
+        <h2>Are you sure you?</h2>
+        <p>
+          Changing the template palette will overwrite any applied styles.
+        </p>
+        <div class="buttons">
+          <paper-button
+            @click=${() => {
+              if (this.paletteConfimationDialog) {
+                this.paletteConfimationDialog.close();
+              }
+              if (this.paletteSelect) {
+                this.paletteSelect.value = this.selectedPalette;
+                this.requestedPalette = this.selectedPalette;
+              }
```

Comment:
Can we use `clearSelectedPalette()` instead?

Reply:
ClearSelectedPalette reverts to the original state. I created another callback for the decline action. 

```Typescript
  /**
   * Called when users decline a palette change.
   */
  paletteChangeDecline() {
    if (this.paletteConfimationDialog) {
      this.paletteConfimationDialog.close();
    }
    if (this.paletteSelect) {
      this.paletteSelect.value = this.selectedPalette;
      this.requestedPalette = this.selectedPalette;
    }
  }
```